### ISSUE TITLE
Using Campus Energy Modeling

### ISSUE THREAD

**Issue by frateanadrian** [NREL/CampusEnergyModeling#1]
Hello, I am trying to use these tools but I am having some issues. Does anyone can provide me some help?

**Comment by stephen-frank** [comment]
We have very limited support capability (this project doesn't have an external support budget), but would be happy to answer any questions we can. What specific questions do you have? Have you followed the installation guide on the [Wiki](https://github.com/NREL/CampusEnergyModeling/wiki/Installation), including adherence to the software versions listed? This project is frozen in time and hasn't yet been updated for compatibility with recent releases of MATLAB etc., so failure to use the exact software versions listed will likely cause problems.

**Comment by frateanadrian** [comment]
Hello,

Thanks for your answer. I am currently working on my PhD Thesis and I want to use MLE+ to simulate building energy consumption with EnergyPlus and build control strategies and other simulations with Simulink and this is how I've discovered this project.
I am running Matlab 2013a and EnergyPlus 8-0-0-0 on a Mac OS 10.11.6.
I downloaded the latest release and I've followed the steps presented in the Wiki.
First I've tried to run de demos, but I get an error saying that there is a bad link for the Simulink Library models.
<img width="547" alt="screen shot 2017-01-24 at 22 22 13" src="https://cloud.githubusercontent.com/assets/8103983/22265182/98b21ca4-e283-11e6-8ef2-935954c4b0bb.png">
What might be the problem? The Simulink Library seems to be a *mdl file, but I am unable to open it with Simulink. Did I miss anything in the process?
I've tried also to use the legacy MLE+ block, but that does't seem to work either. I am only able to use the GUI interface for MLE+.

**Comment by stephen-frank** [comment]
Those "Bad Link" boxes imply that the Campus Energy Modeling library is not on your Simulink path. Did you definitely edit and run the `install.m` script in MATLAB per the installation instructions? If not, try that. If you already did, then maybe something is preventing the script from adding the model library to your path, or else that the path is not persisting after closing MATLAB. You may need to manually add the location of the "Library" directory distributed with the release to your MATLAB path.

**Comment by frateanadrian** [comment]
Yeah I did it. I ttached some screenshots with an error I get when I open a demo file. You can see above that the install.m is done successfully.
<img width="554" alt="screen shot 2017-01-24 at 23 27 45" src="https://cloud.githubusercontent.com/assets/8103983/22267632/3c25f866-e28d-11e6-9dd3-f4538157ce0f.png">
This is the path in one of the blocks:
<img width="664" alt="screen shot 2017-01-24 at 23 28 17" src="https://cloud.githubusercontent.com/assets/8103983/22267656/58c6731a-e28d-11e6-9b97-1447b9d868ff.png">
I tried to add the rest:
<img width="671" alt="screen shot 2017-01-24 at 23 30 58" src="https://cloud.githubusercontent.com/assets/8103983/22267677/6bc8fafa-e28d-11e6-8c27-036a699ec822.png">

**Comment by stephen-frank** [comment]
Can you verify that the path to the library is actually on your MATLAB path? E.G. by running the MATLAB `path` command? If it is, then I'm not quite sure why it would be failing to find it.

**Comment by stephen-frank** [comment]
@willybernal Any ideas? He's running this on a Mac. Maybe permissions errors?

**Comment by frateanadrian** [comment]
Yeah, it is.
<img width="765" alt="screen shot 2017-01-24 at 23 49 00" src="https://cloud.githubusercontent.com/assets/8103983/22268318/c1ba425a-e28f-11e6-8bab-fa328b45a159.png">

**Comment by frateanadrian** [comment]
If I run the test script, this is what I get
<img width="1001" alt="screen shot 2017-01-24 at 23 52 23" src="https://cloud.githubusercontent.com/assets/8103983/22268430/37b757e0-e290-11e6-9e6f-6a0f467e1078.png">

**Comment by stephen-frank** [comment]
Yeah, ok, it definitely seems to be having trouble loading the actual Simulink library, despite having it on the path. Unfortunately, the errors you are seeing are rather cryptic. I am not a Mac user, but it may have something to do with Macintosh permissions. (I'm taking a shot in the dark here.) My colleague Willy Bernal uses a Mac. I will see if he has any ideas, or can perhaps refer you to a different strategy for using MLE+.

**Comment by willybernal** [comment]
Hello Adrian,

I am not sure why you are experiencing this problem. As you are mainly interested in MLE+, I would try to get that piece running properly and then we can revisit how to make it work with the CampusEnergyModeling toolbox:

- Install MLE+ from here [https://github.com/willybernal/mlep](https://github.com/willybernal/mlep)

- Follow the installation instructions -> [https://github.com/willybernal/mlep/wiki/Installation](https://github.com/willybernal/mlep/wiki/Installation)
- This installation has been tested with multiple Matlab versions and should be good to go. MLE+ can be used with multiple E+ versions as well (listed in the website). However, the example files, specifically the E+ files, will only work with E+ 8.6.0.

- Try the Legacy Example. I haven't updated more example files yet.

Let me know if you get this working or if you have any questions. After that, I can help you out getting the Campus EnergyModeling toolbox working.

Thanks,
Willy

**Comment by frateanadrian** [comment]
Hi Willy,

Thank you for answering.
I tried the legacy example, but I get this error:
<img width="732" alt="screen shot 2017-01-25 at 23 59 39" src="https://cloud.githubusercontent.com/assets/8103983/22311092/864f9320-e35a-11e6-8f67-60406f3afdb5.png">

It seems like there is an issue with Java.

Adrian

**Comment by willybernal** [comment]
I see. Can you open the .err file and send it back? It should be called
simple.err.

Willy Bernal
Mechatronics and Controls Engineer
willybernal@gmail.com

**Comment by willybernal** [comment]
Make sure you have E+ 8.6 and you have changed the settings in
installMlep.m before running that file.

Willy Bernal
Mechatronics and Controls Engineer
willybernal@gmail.com

**Comment by frateanadrian** [comment]
I have 8.6 and I changed the path. You mean err file form Matlab?

**Comment by willybernal** [comment]
The .err file gets generated by E+. That should have information about the
error you are encountering. The error is showing is just saying that E+ did
not respond on time to Matlab. So, the socket connection opened in Matlab
times out and then throws that error. There must be some problem with the
settings. That information probably shows up in the .err file.

I actually meant to say SmOffPSZ.err. You should take a look at that to see
what would be the problem. Also, send it my way please.

Willy Bernal
Mechatronics and Controls Engineer
willybernal@gmail.com

**Comment by frateanadrian** [comment]
It's strange, I've been searching everywhere but there is no .err file that I get after trying to run the simulation.

**Comment by willybernal** [comment]
This is what my legacy folder looks like:
It might be inside the Output folder. Can you send me a screenshot of your
folder after you try to run simpl.slx?

Willy Bernal
Mechatronics and Controls Engineer
willybernal@gmail.com

**Comment by frateanadrian** [comment]
this is how it looks like after the simulation:
<img width="326" alt="screen shot 2017-01-27 at 21 42 04" src="https://cloud.githubusercontent.com/assets/8103983/22385024/a08aa736-e4d9-11e6-85b4-40280308321b.png">
i have no output folder

**Comment by frateanadrian** [comment]
i think you missed adding a pic of your folder
